pkg/build/controller/buildpod/factory.go

```diff
@@ -0,0 +1,213 @@
+package buildpod
```

Comment:
It is common in go to have methods of a single struct span multiple files.
But I don't feel strongly one way or the other, so will collapse.